packages/sales-api-service/src/services/__tests__/concession.service.spec.js

```diff
@@ -0,0 +1,283 @@
+    name: CONCESSION.SENIOR
+  },
+  {
+    id: 'eee-555-fff-666',
+    name: CONCESSION.DISABLED
+  }
+])
+
+getReferenceDataForEntityAndId.mockImplementation((_e, id) => {
+  if (id === 'aaa-111-bbb-222') {
+    return {
+      id,
+      name: CONCESSION.JUNIOR
+    }
+  }
+  if (id === 'ccc-333-ddd-444') {
+    return {
+      id,
+      name: CONCESSION.SENIOR
+    }
+  }
+  if (id === 'eee-555-fff-666') {
+    return {
+      id,
+      name: CONCESSION.DISABLED
+    }
+  }
+  throw Error('Unrecognised concession id')
+})
+
+const getMockPermission = () => ({})
+
+const getJuniorConcession = () => ({
+  id: 'aaa-111-bbb-222',
+  name: CONCESSION.JUNIOR,
+  proof: {
+    type: CONCESSION_PROOF.none
+  }
+})
+
+const getSeniorConcession = () => ({
+  id: 'ccc-333-ddd-444',
+  name: CONCESSION.SENIOR,
+  proof: {
+    type: CONCESSION_PROOF.none
+  }
+})
+
+const getDisabledBlueBadgeConcession = () => ({
+  id: 'eee-555-fff-666',
+  name: CONCESSION.DISABLED,
+  proof: {
+    type: CONCESSION_PROOF.blueBadge,
+    referenceNumber: 'blue-badge-123'
+  }
+})
+
+const getDisabledNiConcession = () => ({
+  id: 'eee-555-fff-666',
+  name: CONCESSION.DISABLED,
+  proof: {
+    type: CONCESSION_PROOF.NI,
+    referenceNumber: 'national-insurance-456'
+  }
+})
+
+describe('preparePermissionDataForRenewal', () => {
+  describe('Disabled', () => {
+    describe('addDisabled', () => {
+      it('add blue badge', async () => {
+        const permission = getMockPermission()
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.blueBadge, 'bb-111')
+        expect(permission.concessions).toContainEqual(
+          expect.objectContaining({
+            id: 'eee-555-fff-666',
+            name: CONCESSION.DISABLED,
+            proof: {
+              type: CONCESSION_PROOF.blueBadge,
+              referenceNumber: 'bb-111'
+            }
+          })
+        )
+      })
+
+      it('add NI', async () => {
+        const permission = getMockPermission()
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.NI, 'national-insurance-456')
+        expect(permission.concessions).toContainEqual(expect.objectContaining(getDisabledNiConcession()))
+      })
+
+      it('add blue badge to senior licence', async () => {
+        const permission = getMockPermission()
+        permission.concessions = [getSeniorConcession()]
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.blueBadge, 'blue-badge-123')
+        expect(permission.concessions).toContainEqual(getDisabledBlueBadgeConcession())
+      })
+
+      it('add NI to senior licence', async () => {
+        const permission = getMockPermission()
+        permission.concessions = [getSeniorConcession()]
+        await concessionService.addDisabled(permission, CONCESSION_PROOF.NI, 'national-insurance-456')
+        expect(permission.concessions).toContainEqual(getDisabledNiConcession())
+      })
+    })
+
+    describe('removeDisabled', () => {
```

Comment:
Do we need to test that any other concessions are not also removed (so if the permission also has a senior concession and a disabled concession, removing the disabled one does not affect the senior)? Same goes for tests on the other `remove` functions.

Reply:
Weirdly, I'd already done this for other concessions, just not for disabled concessions